lib/active_model/serializer.rb

```diff
@@ -154,6 +154,16 @@ def links
       self.class._links
     end
 
+    # @api private
+    # Returns an ActiveModel::Name for the serializer's object
+    def model_name
```

Comment:
This is a really interesting change. Would you mind adding some tests to show how serializing an object without a `model_name` behaves?